### PR TITLE
Convert osg-incommon-cert-request to use urllib3, to fix for el9

### PR DESCRIPTION
**Pull request by DrDaveD**
This changes osg-incommon-cert-request to use urllib3 instead of M2Crypto, to make it work on el9.

### CODE REVIEW

File: osgpkitools/cert_utils.py [hunk 1]
```diff
@@ -18,35 +23,18 @@
 
 
 def get_ssl_context(usercert, userkey):
-    """ This function sets the ssl context by accepting the passphrase
-    and validating it for user private key and his certificate
-    INPUT
+    """ This function sets the ssl context for urllib3.
+        It will prompt for password if needed.
```
Comment (matyasselmeci) [inline osgpkitools/cert_utils.py:27]:
```suggestion
        OpenSSL will prompt for password if needed.
```

**Review by matyasselmeci — COMMENTED** [review]
I have a few comments; mostly fine though.

File: osgpkitools/cert_utils.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 
-import getpass
 import os
 import sys
 import tempfile
 
-from M2Crypto import SSL, m2, RSA, EVP, X509
+try:
+    from urllib3.util import create_urllib3_context
+except:
```
[the remark this answers is not in the document]
Comment (DrDaveD) [reply osgpkitools/cert_utils.py]: Ok

File: osgpkitools/rest_client.py [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
 from .ExceptionDefinitions import *
 
+prog = "osg-incommon-cert-request"
+
 logger = logging.getLogger(__name__)
 
 class InCommonApiClient():
 
-    def __init__(self, base_url, ssl_context, *args, **kwargs):
+    def __init__(self, base_url, sslcontext, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (DrDaveD) [reply osgpkitools/rest_client.py]: I changed it to distinguish the variable name from the parameter used in the call to the PoolManager, but I guess that's not the typical way to do things, so I changed it back.

File: osgpkitools/rest_client.py [hunk 4]
```diff
@@ -1,32 +1,37 @@
...
         """base_url and api_timeout
 
         Args:
             base_url (string): Will be used to connect to the InCommon API (cert-auth) 
+            clientargs (object): Arguments including usercert and userprivkey
```
[the remark this answers is not in the document]
Comment (DrDaveD) [reply osgpkitools/rest_client.py]: Oh right.  I had clientargs in there in an intermediate development that I threw away, and forgot to change it back.

File: osgpkitools/cert_utils.py (hunk 1 above)
Comment (DrDaveD) [reply osgpkitools/cert_utils.py]: Ok

**Review by matyasselmeci — APPROVED** [review]

**Pull request merged**